bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/SearchResultsStorefrontContextImpl.java

```diff
@@ -0,0 +1,131 @@
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.internal.storefrontcontext;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultCategory;
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultProduct;
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultSuggestion;
+import com.adobe.cq.commerce.core.components.storefrontcontext.SearchResultsStorefrontContext;
+import com.adobe.cq.commerce.core.search.models.SearchAggregation;
+import com.adobe.cq.commerce.core.search.models.SearchResultsSet;
+
+public class SearchResultsStorefrontContextImpl extends AbstractCommerceStorefrontContext implements SearchResultsStorefrontContext {
+
+    private final SearchResultsSet searchResultsSet;
+
+    public SearchResultsStorefrontContextImpl(SearchResultsSet searchResultsSet) {
+        this.searchResultsSet = searchResultsSet;
+    }
+
+    @Override
+    public String getSearchUnitId() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getSearchRequestId() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public List<SearchResultProduct> getProducts() {
+        return searchResultsSet.getProductListItems().stream()
+            .map(p -> new SearchResultProduct() {
+
+                @Override
+                public String getName() {
+                    return p.getTitle();
+                }
+
+                @Override
+                public String getSku() {
+                    return p.getSKU();
+                }
+
+                @Override
+                public String getUrl() {
+                    return p.getURL();
+                }
+
+                @Override
+                public String getImageUrl() {
+                    return p.getImageURL();
+                }
+
+                @Override
+                public Double getPrice() {
+                    return p.getPriceRange().getFinalPrice();
+                }
+
+                @Override
+                public int getRank() {
+                    return 0;
+                }
+            }).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<SearchResultCategory> getCategories() {
+        SearchAggregation searchAggregationOptions = searchResultsSet
+            .getSearchAggregations().stream()
+            .filter(a -> a.getIdentifier().equals("category_id"))
+            .findFirst()
+            .orElse(null);
+
+        if (searchAggregationOptions != null) {
+            return searchAggregationOptions.getOptions().stream()
+                .map(o -> new SearchResultCategory() {
```

Comment:
Same here, please use static nested classes when you don't need a reference to the outer classes instance

Reply:
https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/searchResults.ts
I am replicating these types. these need to be available in order to be used with other methods of the MSE api